extractor/src/main/scala/com/digitalasset/extractor/Extractor.scala

```diff
@@ -47,7 +47,7 @@ class Extractor[T](config: ExtractorConfig, target: T)(
   private val tokenHolder = config.accessTokenFile.map(new TokenHolder(_))
   private val parties: Set[String] = config.parties.toSet.map(identity)
 
-  implicit val system: ActorSystem = ActorSystem()
+  lazy implicit val system: ActorSystem = ActorSystem()
```

Comment:
> Make ActorSystem lazy so subclasses can override to customize it.

I'm not sure I understand how laziness affects the possibility of overriding this, can you explain, please?

Reply:
It affects whether references to `system` can be overridden _in time_.

```scala
scala> class Foo {val x = ""; val y = x}; class Bar extends Foo {override val x = "please be y"}; (new Bar).y
class Foo
class Bar
val res1: String = null

scala> class Foo {lazy val x = ""; val y = x}; class Bar extends Foo {override lazy val x = "please be y"}; (new Bar).y
class Foo
class Bar
val res3: String = please be y
```

Reply:
I had found an interesting way to override it, but the base class's value is still initialized before being overridden and that  value is not referenceable.
So in this case, an extra `ActorSystem` is created and dosn't shutdown.
```scala
class Foo {val x = ""; val y = x}; class Bar(override val x: String = "please be y") extends Foo {}; (new Bar).y
```